### PR TITLE
Use the org.jboss.redhat-fuse BOM in the quickstarts

### PR DESCRIPTION
**Pull request by cunningt**
I took a first stab at converting over the quickstarts, was hoping to get someone to look it over to make sure this is the right way to do it.    I've made a number of changes here : 

- use org.jboss.redhat-fuse BOM in each of the quickstarts
- removed the parent pom - the quickstarts are now parentless and could be copied and used by developers as sample projects

This is a first pass - there's a number of things here that still need to be done 

- custom quickstart still needs a cxf version and a fuse-karaf version because of the fuse-branding module.   @grgrzybek is there a way around that?    I'm not exactly sure what this quickstart does - do we need to include the fuse-branding in this one?
- ~cxf/rest is failing in https://issues.jboss.org/browse/ENTESB-9637~ (fixed in https://github.com/jboss-fuse/redhat-fuse/pull/29)
- ~everything needs to be run with tests~ - fixed this by adding qtest profile to all the quickstarts, did a quick run through of the quickstart tests with a kit
- need to figure out whether `<repositories/>` are necessary

### CODE REVIEW

**Comment by RickJWagner** [comment]
+1 for parentless poms in the quickstarts.  Thanks for that.  (Makes it easier, you don't have to put your own examples in the quickstarts directory for them to work.)

**Comment by grgrzybek** [comment]
I'll have a look at it. I generally like the idea of parentless (orphaned?) quickstarts - I didn't check it but is `<pluginManagement>` imported with BOM as `<dependencyManagement>` is?

**Comment by cunningt** [comment]
@grgrzybek My understanding is that `<pluginManagement>` is not imported with a BOM like `<dependencyManagement>` is.

Not sure you are aware, but we are repackaging the maven plugins that we build as org.jboss.redhat-fuse (https://github.com/jboss-fuse/redhat-fuse/tree/master/fuse-maven-plugins) - if you use that abstraction (i.e. org.jboss.redhat-fuse:camel-maven-plugin rather than org.apache.camel:camel-maven-plugin) you get the version setup for you if you import the redhat-fuse BOM.

Where I'm running into trouble and could use your help is the `custom` quickstart.    The goal here is that the only property we have in the pom is `<fuse.version>`, you import the BOM, and you get all of your Fuse dependencies through that - but the custom quickstart requires a mess of properties (fuse-karaf version, karaf version, servicemix-specs, etc) - is there a way we can simplify this?

File: pom.xml [hunk 1]
```diff
@@ -22,323 +22,34 @@
...
-                    <obrRepository>NONE</obrRepository>
-                    <instructions>
-                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
-                    </instructions>
-                </configuration>
-            </plugin>
-        </plugins>
-    </build>
-
     <modules>
```
Comment (valdar) [inline pom.xml:30]: Why explicitly listing the sub sub modules?

Comment (cunningt) [reply pom.xml:30]: My thought here is this 

- we don't want the quickstarts to have parents - if you remove parentage from the quickstarts, then developers can grab them quickly and adapt them to whatever scenario they want
- we do want to build the quickstarts all in one bunch

Comment (grgrzybek) [reply pom.xml:30]: We can build each of them except custom-distro

**Comment by grgrzybek** [comment]
I've _flattened_ all the quickstarts in 7.x.redhat-7-x branch

**Pull request closed without merging**